lib/compiler-cranelift/src/translator/code_translator.rs

```diff
@@ -1894,7 +2062,7 @@ fn mem_op_size(opcode: ir::Opcode, ty: Type) -> u32 {
         ir::Opcode::Istore8 | ir::Opcode::Sload8 | ir::Opcode::Uload8 => 1,
         ir::Opcode::Istore16 | ir::Opcode::Sload16 | ir::Opcode::Uload16 => 2,
         ir::Opcode::Istore32 | ir::Opcode::Sload32 | ir::Opcode::Uload32 => 4,
-        ir::Opcode::Store | ir::Opcode::Load => ty.bytes(),
+        ir::Opcode::Store | ir::Opcode::Load /* | ir::Opcode::LoadSplat */=> ty.bytes(),
```

Comment:
I think this would be wrong for LoadSplat -- LoadSplat loads a small size (1 byte / 2 byte / 4 byte / 8 byte) and splats it to the full 16 byte. If you use the type of the value pushed onto the stack by the loadsplat instruction then you would always return 16 for the V128. The function is supposed to return how many bytes it loads/stores.

Reply:
It's directly ported from cranelift-wasm.